devicetypes/fibargroup/fibaro-door-window-sensor-2.src/fibaro-door-window-sensor-2.groovy

```diff
@@ -129,12 +131,26 @@ def updated() {
 	}
 	logging("${device.displayName} - Executing updated()","debug")
 
-	if ( settings.temperatureHigh as Integer == 0 && settings.temperatureLow as Integer == 0 ) { 
-		sendEvent(name: "temperatureAlarm", value: null, displayed: false) 
-	} else if ( settings.temperatureHigh != null || settings.temperatureHigh != null ) {
-		sendEvent(name: "temperatureAlarm", value: "clear", displayed: false) 
+	def tempAlarmMap = ["clear": "cleared", "overheat": "heat", "underheat": "freeze"]
+	// Convert old device specific temperatureAlarm event to standard Temperature Alarm capability event
+	def temperatureAlarmCurrentValue = device.currentValue("temperatureAlarm")
+	if (tempAlarmMap.containsKey(temperatureAlarmCurrentValue)) {
+		sendEvent(name: "temperatureAlarm", value: tempAlarmMap[temperatureAlarmCurrentValue], displayed: true)
 	}
-	
+
+	def currentTemperature = device.currentValue("temperature")
+	def alarmCleared = device.currentValue("temperatureAlarm") == "cleared"
+	def alarmFreeze = device.currentValue("temperatureAlarm") == "freeze"
+	def alarmHeat = device.currentValue("temperatureAlarm") == "heat"
+	def temperatureHigh = (settings.temperatureHigh ? new BigDecimal(settings.temperatureHigh) : null) * 0.1
+	def temperatureLow = (settings.temperatureLow ? new BigDecimal(settings.temperatureLow) : null) * 0.1
+	if (!alarmCleared) {
+		if ((temperatureHigh != null && (currentTemperature < temperatureHigh) && !alarmFreeze) ||
+			(temperatureLow != null && (currentTemperature > temperatureLow) && !alarmHeat)) {

```

Comment:
another reason to be using a default value is that this should be an AND condition, right?

If temp is below max temp & temp is above min temp, send clear

Reply:
These thresholds can be changed independently so it is better to check them separately as well.

Reply:
right, but if the temperature is below the max temperature this or statement will short-circuit and send a clear, even if it's also below the minimum temperature, right?

Reply:
no, it won't, because if the temperature is below max temperature and at the same time below a minimum temperature, forcing cleared event will be blocked by the AND !alarmFreeze